src/inspector_agent.cc

```diff
@@ -62,6 +64,21 @@ bool AcceptsConnection(InspectorSocket* socket, const std::string& path) {
                             sizeof(DEVTOOLS_PATH) - 1);
 }
 
+std::string FormatJSON(const std::map<std::string, std::string> object) {
+  std::ostringstream json;
+  json << "[ {\n";
+  bool first = true;
+  for (const auto& name_value : object) {
+    if (!first)
+      json << ",\n";
+    json << "  \"" << name_value.first << "\": \"";
+    json << name_value.second << "\"";
+    first = false;
+  }
+  json << "\n} ]";
+  return json.str();
+}
```

Comment:
A homegrown JSON serializer seems like a recipe for disaster.  Is there a way to use V8's JSON.stringify?

If nothing else, `FormatJSON()` is something of a misnomer because you need to escape the values manually.